test/test_custom_ops.py

```diff
@@ -2969,6 +2969,26 @@ def numpy_sin_inplace(x: Tensor) -> Tensor:
         with self.assertRaisesRegex(RuntimeError, "may not alias"):
             numpy_sin_inplace(x)
 
+    def test_factory_function(self):
+        @torch.library.custom_op(
+            "_torch_testing::f", mutates_args={}, device_types="cpu"
+        )
+        def f(device: torch.device) -> Tensor:
+            return torch.ones(3)
+
+        result = f(device="cpu")
+        self.assertEqual(result.device, torch.device("cpu"))
+        self.assertEqual(result, torch.ones(3))
```

Comment:
Can you test the case where we split registrations too?

```
@torch.library.custom_op(
    "_torch_testing::f", mutates_args={}, device_types="cpu"
 )
 def f() -> Tensor:
     raise NotImplementedError("NYI")

@f.register_kernel("cpu")
def _():
    return torch.zeros(3)

result = f(x)
```

This case should error out somewhere saying that we need a device: Device keyword arg

Reply:
Sorry, I meant the following:
```
@torch.library.custom_op(
    "_torch_testing::f", mutates_args={}
 )
 def f() -> Tensor:
     raise NotImplementedError("NYI")

@f.register_kernel("cpu")
def _():
    return torch.zeros(3)

result = f(x)
```

It should be fine to register a factory function that doesn't have a `device: Device` arg if there is no separate backend dispatch. So, the following should be OK:

```
@torch.library.custom_op(
    "_torch_testing::f", mutates_args={}
 )
 def f() -> Tensor:
     return torch.ones(3)
```

Reply:
makes sense. fixed now.